src/core/meshcop/leader.cpp

```diff
@@ -31,6 +31,8 @@
  *   This file implements a MeshCoP Leader.
  */
 
+#if OPENTHREAD_FTD
```

Comment:
Not really - i was not very consistent only due to the quantity of these I had to add. 

Fundamentally this is sort of "configure" option, but - we can only have a single configure file and build two different library configurations (ie: MTD and FTD)

I think it is fine because this is a *commandline* define, and not done in a config.h file